engine/src/main/java/org/terasology/logic/common/ActivateEvent.java

```diff
@@ -84,16 +84,24 @@ public int getActivationId() {
 
     public Vector3f getTargetLocation() {
         LocationComponent loc = target.getComponent(LocationComponent.class);
-        if (loc != null&& !Float.isNaN(loc.getWorldPosition().x)) {
-            return loc.getWorldPosition();
+        if (loc != null) {
+            Vector3f result = loc.getWorldPosition(new Vector3f());
+            if (result.isFinite()) {
+                return result;
+            }
         }
         return null;
     }
 
     public Vector3f getInstigatorLocation() {
         LocationComponent loc = instigator.getComponent(LocationComponent.class);
-        if (loc != null&& !Float.isNaN(loc.getWorldPosition().x)) {
-            return loc.getWorldPosition();
+        if (loc != null) {
+            Vector3f result = loc.getWorldPosition(new Vector3f());
+            if (result.isFinite()) {
+                return result;
+            }
+            result.set(0, 0, 0);
+            return result;
```

Comment:
Why do we return [0,0,0] vector here but `null` above in `getTargetLocation()`?

Reply:
I have no idea why we do this. It's kind of strange. Don't want to think about it too much. We can look at it in another pass.

Reply:
Okay, different question: Do we need this at all if we have the `return new Vector3f()` at the end?
```
            result.set(0, 0, 0);
            return result;
```

Reply:
Yeah I see that, but isn't a cleared vector equal to an empty vector?
So the code path leads to the return of the empty vector (`return new Vector3f()` in l.106) anyway, so there's no need to explicitly clear and return at that point, is there?

Reply:
I'm assuming there is some math that will add it or something and a nan vector will throw a wrench into things. Better to clear it. Basically the same as passing out a new vector.

Reply:
I see okay. Fair point.